### ISSUE TITLE
Path/baseUrl creates broken Type Definitions

### ISSUE THREAD

**Issue by dolanmiu** [Microsoft/TypeScript#21507]
**TypeScript Version:**  2.6.2

**Search Terms:** npm, link, declarations, type definitions, awesome-typescript-loader, paths, baseUrl

I have tried to search for this, but couldn't find something similar, so I am asking 😄.

I am using `awesome-typescript-loader` to bundle my typescript project into one `build/index.js` file:
https://github.com/dolanmiu/docx/blob/master/webpack.config.js#L7-L8

I am using `compilerOptions.paths` to enable me to import from root:
https://github.com/dolanmiu/docx/blob/master/tsconfig.json#L14-L18

I also have `declarations: true`:
https://github.com/dolanmiu/docx/blob/master/tsconfig.json#L11

However, when I `npm link`, and use the library, I get the following error:
- Notice how it is complainig about the root imports, such as `file/media`
- `packer` is an interface

```sh
../../docx/build/export/index.d.ts(3,15): error TS2307: Cannot find module './packer/packer'.
../../docx/build/export/packer/express.d.ts(3,22): error TS2307: Cannot find module 'file'.
../../docx/build/export/packer/express.d.ts(4,25): error TS2307: Cannot find module './packer'.
../../docx/build/export/packer/local.d.ts(2,25): error TS2307: Cannot find module './packer'.
../../docx/build/file/document/body/section-properties/columns/columns-attributes.d.ts(1,39): error TS2307: Cannot find module 'file/xml-components'.
../../docx/build/file/document/body/section-properties/doc-grid/doc-grid-attributes.d.ts(1,39): error TS2307: Cannot find module 'file/xml-components'.
../../docx/build/file/document/body/section-properties/page-margin/page-margin-attributes.d.ts(1,39): error TS2307: Cannot find module 'file/xml-components'.
../../docx/build/file/document/body/section-properties/page-size/page-size-attributes.d.ts(1,39): error TS2307: Cannot find module 'file/xml-components'.
../../docx/build/file/document/body/section-properties/section-properties.d.ts(1,30): error TS2307: Cannot find module 'file/xml-components'.
../../docx/build/file/document/document.d.ts(1,28): error TS2307: Cannot find module 'file/media'.
../../docx/build/file/document/document.d.ts(2,30): error TS2307: Cannot find module 'file/xml-components'.
```

![image](https://user-images.githubusercontent.com/2917613/35603241-42ef511c-0634-11e8-823c-2faa6412d963.png)

**Expected behavior:** Create typescript definitions without red underline errors

**Actual behavior:** Red underline errors, and error log

**Related Issues:**

https://github.com/Microsoft/TypeScript/issues/6496

Project:

https://github.com/dolanmiu/docx

**Comment by mhegazy** [comment]
I am not sure i understand the intention behind your `paths`. if all what you want to say is resolve `file/media` to `./src/file/media/index`, then you all you need is the `baseUrl` and remove `paths`.

I tried running `tsc` on the tsconfig.json you provided in docx, and i got no module related errors.. i am guessing this is a loader issue. i would recommend you filing it on https://github.com/s-panferov/awesome-typescript-loader/issues instead.

**Comment by dolanmiu** [comment]
I have the same issue when removing `paths` and only keeping the `baseUrl`

I have tried `tsc`, and yes, you are right, it does compile with no errors, but when I try running a demo or using the library, I get the error:

```
Error: Cannot find module 'file/xml-components'
    at Function.Module._resolveFilename (module.js:555:15)
    at Function.Module._load (module.js:482:25)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\Dolan\Documents\docx\build\file\paragraph\formatting\alignment.js:3:26)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
```

This makes sense as `alignment.js` has:

```js
const xml_components_1 = require("file/xml-components");
```

Obviously it won't be able to find `require("file/xml-components")`, cos its specific to the library

**Comment by mhegazy** [comment]
The compiler will not change your file path/import module name. what baseUrl tells the compiler is that at runtime *all* modules will be copied to this folder. AMD modules are usually distributed this way. if this is not your setup, do not use `baseUrl`/`paths`.
